### PR TITLE
[JENKINS-64107] Convert SSHD module to a plugin

### PR DESCRIPTION
**Pull request by kuisathaverat**
[JENKINS-55582](https://issues.jenkins-ci.org/browse/JENKINS-55582) Convert modules to plugins
[JENKINS-64107](https://issues.jenkins-ci.org/browse/JENKINS-64107) Convert SSHD module to a plugin
[JEP](https://github.com/jenkinsci/jep/pull/326) Convert modules to plugins

* This PR change to use the  sshd-plugin
* Bump the Core to the 2.282

### CODE REVIEW

**Comment by kuisathaverat** [comment]
waiting for the core 2.282 to be released

**Comment by jglick** [comment]
We could just delete this whole dep if we cleaned out the old execution modes. See https://github.com/jenkinsci/ssh-agent-plugin/pull/47#issuecomment-795667756

**Comment by jglick** [comment]
Would be obsoleted by #48.

**Comment by jglick** [comment]
That is not merged yet!

File: pom.xml [hunk 1]
```diff
@@ -79,9 +79,9 @@
   <dependencies>
     <!-- regular dependencies -->
     <dependency>
-      <groupId>org.apache.sshd</groupId>
-      <artifactId>sshd-core</artifactId>
-      <version>1.7.0</version>
+      <groupId>org.jenkins-ci.modules</groupId>
+      <artifactId>sshd</artifactId>
+      <version>3.0.1</version>
```
Comment (kuisathaverat) [inline pom.xml:84]:
```suggestion
      <version>3.0.3</version>
```

**Comment by jglick** [comment]
Can be closed now I think @alecharp?

**Pull request closed without merging**